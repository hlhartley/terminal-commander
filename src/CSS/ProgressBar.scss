@import '_variables.scss';

.progress-container {
    display: flex;
    justify-content: space-between;
    .tank-container {
        animation: bobbing .35s infinite ease-in-out alternate;
    }
    @keyframes bobbing {
        0%{
            transform: rotate(0) translateY(0);
        }
        100%{
            transform: rotate(0.1deg) translateY(3px);
        }
    }
    .tank-img {
        height: 100px;
        transition-duration: 2s;
        position: relative;
    }
    .star-img {
        height: 60px;
        margin-right: 12px;
        margin-top: 25px;
    }
    .pulse {
        animation: pulse 3s infinite;
        background: #cbb05375;
        border-radius: 50%;
        box-shadow: 0 0 0 rgba(204,169,44, 0.4);
        cursor: pointer;
        display: block;
        width: 30px;
        height: 30px;
        &:hover {
            animation: none;
        }
    }
    @-webkit-keyframes pulse {
        0% {
            -webkit-box-shadow: 0 0 0 0 rgba(204,169,44, 0.4);
        }
        70% {
            -webkit-box-shadow: 0 0 0 10px rgba(204,169,44, 0);
        }
        100% {
            -webkit-box-shadow: 0 0 0 0 rgba(204,169,44, 0);
        }
    }
    @keyframes pulse {
        0% {
            -moz-box-shadow: 0 0 0 0 rgba(204,169,44, 0.4);
            box-shadow: 0 0 0 0 rgba(204,169,44, 0.4);
        }
        70% {
            -moz-box-shadow: 0 0 0 10px rgba(204,169,44, 0);
            box-shadow: 0 0 0 10px rgba(204,169,44, 0);
        }
        100% {
            -moz-box-shadow: 0 0 0 0 rgba(204,169,44, 0);
            box-shadow: 0 0 0 0 rgba(204,169,44, 0);
        }
    }
}
