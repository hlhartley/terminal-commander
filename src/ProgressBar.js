import React, { Component } from 'react';
import './CSS/main.scss';

export default class ProgressBar extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const transformStyles = {
      transform: `translate(${this.props.score*3}vw)`,
    }

    return (
      <div className='progress-container'>
          <img src='/images/armytank.png' className='tank-img' alt='tank' style={transformStyles}></img>
          <img src='/images/star.png' className='star-img pulse' alt='yellow star'></img>
      </div>
    );
  }
}