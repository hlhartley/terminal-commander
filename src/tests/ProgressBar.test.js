import React from 'react';
import ProgressBar from '../ProgressBar';
import { shallow } from 'enzyme';

let wrapper;
describe('ProgressBar', () => {

  beforeEach(() => {
    wrapper = shallow(
      <ProgressBar />
    );
  })

  it('should exist', () => {  
    expect(wrapper).toBeDefined()
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})