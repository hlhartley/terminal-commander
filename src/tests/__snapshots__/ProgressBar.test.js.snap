// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="tank"
          className="tank-img"
          src="/images/armytank.png"
        />,
        <img
          alt="yellow star"
          className="star-img pulse"
          src="/images/star.png"
        />,
      ],
      "className": "progress-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "tank",
          "className": "tank-img",
          "src": "/images/armytank.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "yellow star",
          "className": "star-img pulse",
          "src": "/images/star.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="tank"
            className="tank-img"
            src="/images/armytank.png"
          />,
          <img
            alt="yellow star"
            className="star-img pulse"
            src="/images/star.png"
          />,
        ],
        "className": "progress-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "tank",
            "className": "tank-img",
            "src": "/images/armytank.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "yellow star",
            "className": "star-img pulse",
            "src": "/images/star.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
